raiden/network/rpc/client.py

```diff
@@ -155,7 +156,12 @@ def format_data_for_call(
 def check_node_connection(func):
     """ A decorator to reconnect if the connection to the node is lost."""
     def retry_on_disconnect(self, *args, **kwargs):
+        """self here is always an obhect of JSONRPCClient"""
         for i, timeout in enumerate(timeout_two_stage(10, 3, 10)):
+
+            if self.shutting_down:
+                raise RaidenShuttingDown()
```

Comment:
This is a really weird place to raise such exception, what about raising this exception from the `stop` in `RaidenService` after a timeout?

Another approach, clearer IMO, is to add an `stop_event` to the `RaidenService` class, this would be set on the `stop` method and used to signal to *all* greenlets that the process is stopping (not necessarily quitting)

Reply:
Regarding the stop event I also thought about it as I was putting the flag. Was not sure where to place it. The raiden service god object could work. Let me play with it.

But how would that work with stopping the loop? The only object accessible from inside the retry loop is the rpc client object. Hmm ... could just pass the stop_event to the RPC client from the raiden service. 

I think that would work fine. What do you think @hackaugusto ?

Reply:
> could just pass the stop_event to the RPC client from the raiden service.

Yeah, this is the down side of this approach, the `stop_event` would be really pervasive through out the code base. Basically anything that is a greenlet would need to get a reference to it.

The upside is that we don't need to regard stopping as an exceptional case (which I don't think it should be anyways) and a bit more consistent IMO

Reply:
between passing the stop_event downwards or throwing an exception from the bottom and trying to catch it wherever we need to I think the first option is indeed best.